### PR TITLE
Try/deprecate promise

### PR DESCRIPTION
**Pull request by Stojdza**
Introducing async/await in wp-e2e-tests

### CODE REVIEW

**Comment by brbrr** [comment]
Before flow would jump into review your changes, don’t you mind to include a checked list(todo vs done) with all the tests which was updated. That will make a functional review a bit easier :)

**Comment by Stojdza** [comment]
@brbrr I started with `/specs` folder and this is current status:
- [ ] wp-invite-users-spec.js
- [ ] wp-log-in-out-spec.js
- [ ] wp-manage-domains-spec.js
- [ ] wp-media-upload-spec.js
- [ ] wp-notifications-spec.js
- [ ] wp-page-editor-spec.js
- [ ] wp-plan-purchase-spec.js
- [ ] wp-post-editor-spec.js
- [ ] wp-reader-spec.js
- [ ] wp-signup-spec.js
- [ ] wp-theme-multisite-spec.js
- [ ] wp-theme-switch-spec.js
- [ ] wp-view-site-spec.js

File: lib/components/shopping-cart-widget-component.js [hunk 1]
```diff
@@ -10,96 +10,40 @@ export default class ShoppingCartWidgetComponent extends BaseContainer {
...
-				}
-			},
-			function() {
-				d.reject( 'Cart not open' );
-			}
-		);
-
-		return d.promise;
+	async displayedOpen() {
+		return await driverHelper.isElementPresent( this.driver, by.css( 'cart-body' ) );
```
Comment (alisterscott) [inline lib/components/shopping-cart-widget-component.js]: I'm not sure the fullfill and reject etc. above are needed at all - maybe a separate PR to remove these without moving to await just yet to see if this will just work?!?

File: lib/components/shopping-cart-widget-component.js [hunk 2]
```diff
@@ -10,96 +10,40 @@ export default class ShoppingCartWidgetComponent extends BaseContainer {
...
 	}
 
-	removeItem( self ) {
-		let d = webdriver.promise.defer();
```
Comment (alisterscott) [inline lib/components/shopping-cart-widget-component.js]: Again, this may not be needed - may be able to remove this in a separate PR

Comment (Stojdza) [reply lib/components/shopping-cart-widget-component.js]: Sure, I'll make separate PR for this.

**Comment by Stojdza** [comment]
I will close this PR since tests are not passing and it will be split into several smaller PRs.
